### ISSUE TITLE
Deprecation of android.text.format.Time

### ISSUE THREAD

**Issue by frankFic** [dmfs/opentasks#467]
Hi,
I found that class android.text.format.Time is used in many places in open tasks (like [NotificationActionUtils](https://github.com/dmfs/opentasks/blob/6d5db7bc045e4a74cedbea4461719637714a4897/opentasks/src/main/java/org/dmfs/tasks/notification/NotificationActionUtils.java#398), [NotificationUpdaterService](https://github.com/dmfs/opentasks/blob/6d5db7bc045e4a74cedbea4461719637714a4897/opentasks/src/main/java/org/dmfs/tasks/notification/NotificationUpdaterService.java#L588), [TimeRangeCursorFactory](https://github.com/dmfs/opentasks/blob/7eb211a910cd607bb04e93ab9270813c38e9bc9a/opentasks/src/main/java/org/dmfs/tasks/groupings/cursorloaders/TimeRangeCursorFactory.java#L117) and many other places). This class has been deprecated since API since API 22 due to many known issues, including 2038 problem, timezone problem etc. (details in the [doc](https://developer.android.com/reference/android/text/format/Time.html)) As suggested by the doc, [GregorianCalendar](https://developer.android.com/reference/java/util/GregorianCalendar.html) will be a good alternative for Time.

**Comment by dmfs** [comment]
Right, we already started to get rid of this. The provider doesn't use `Time` anymore. We've replaced it with our own `DateTime` implementation and we intend to do the same in the UI module. We'll probably do this when we start to integrate our new date-time picker.
